src/pages/home/report/ReportActionItemMessageEdit.js

```diff
@@ -74,330 +70,303 @@ const defaultProps = {
-            draftMessage = parser.htmlToMarkdown(this.props.draftMessage).trim();
-        } else {
-            // We need to decode saved draft message because it's escaped before saving.
-            draftMessage = Str.htmlDecode(this.props.draftMessage);
+            return parser.htmlToMarkdown(props.draftMessage).trim();
         }
+        // We need to decode saved draft message because it's escaped before saving.
+        return Str.htmlDecode(props.draftMessage);
+    });
+    const [selection, setSelection] = useState({start: 0, end: 0});
+    const [isFocused, setIsFocused] = useState(false);
+    const [hasExceededMaxCommentLength, setHasExceededMaxCommentLength] = useState(false);
 
-        this.state = {
-            draft: draftMessage,
-            selection: {
-                start: 0,
-                end: 0,
-            },
-            isFocused: false,
-            hasExceededMaxCommentLength: false,
-        };
-    }
+    const textInputRef = useRef(null);
+    const wasPreviouslyFocused = usePrevious(isFocused);
 
-    componentDidMount() {
+    useEffect(() => {
         // For mobile Safari, updating the selection prop on an unfocused input will cause it to automatically gain focus
         // and subsequent programmatic focus shifts (e.g., modal focus trap) to show the blue frame (:focus-visible style),
         // so we need to ensure that it is only updated after focus.
-        this.setState((prevState) => ({
-            selection: {
-                start: prevState.draft.length,
-                end: prevState.draft.length,
-            },
-        }));
-    }
-
-    componentWillUnmount() {
-        // Skip if this is not the focused message so the other edit composer stays focused.
-        if (!this.state.isFocused) {
-            return;
-        }
+        setDraft((prevDraft) => {
+            setSelection({
+                start: prevDraft.length,
+                end: prevDraft.length,
+            });
+            return prevDraft;
+        });
 
-        // Show the main composer when the focused message is deleted from another client
-        // to prevent the main composer stays hidden until we swtich to another chat.
-        ComposerActions.setShouldShowComposeInput(true);
-    }
+        return () => {
+            // Skip if this is not the focused message so the other edit composer stays focused
+            if (wasPreviouslyFocused) {
+                return;
+            }
 
-    /**
-     * Update Selection on change cursor position.
-     *
-     * @param {Event} e
-     */
-    onSelectionChange(e) {
-        this.setState({selection: e.nativeEvent.selection});
-    }
+            // Show the main composer when the focused message is deleted from another client
+            // to prevent the main composer stays hidden until we swtich to another chat.
+            ComposerActions.setShouldShowComposeInput(true);
+        };
+        // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
I _think_ we're okay here, but honestly not 100% sure.

Reply:
I think I'm just thrown off because eslint knows that if you do this:

```
const myRef = useRef(1);
useEffect(() => {
   console.log(`Do something with ${myRef.current}`);
}, []);
```

Then you don't have to pass `myRef` or `myRef.current` as a dependency to the `useEffect`. That's exactly what we're doing here, but eslint is thrown off because it doesn't know what `usePrevious` is doing and doesn't know it's just returning `aRef.current` under the hood.

Reply:
Thanks for your thorough re-testing, it's a bummer that `usePrevious` has this limitation. I imagine it would be a useful hook to build into the core React 🤔